crates/project/src/dap_store.rs

```diff
@@ -967,14 +967,44 @@ impl DapStore {
         })
     }
 
+    fn request_dap<R: dap::requests::Request>(
+        &self,
+        client_id: &DebugAdapterClientId,
+        arguments: R::Arguments,
+        cx: &mut ModelContext<Self>,
+    ) -> Task<Result<R::Response>>
+    where
+        <R as dap::requests::Request>::Response: 'static,
```

Comment:
Do we have to specify the lifetime here? Not sure if this is really needed.

Reply:
Without it the compiler returns an error when we try to build.

I think it's because we're moving that type into it's own thread and it needs to have ownership over it's fields to guarantee that they're still exist in the Task.